parser/flux.peg

```diff
@@ -75,10 +75,10 @@ MemberExpressions
 
 MemberExpressionProperty
   = "." __ property:Identifier {
-      return property, nil
+      return membexpr(property)
     }
     / "[" __ property:Primary __ "]" __ {
-      return property, nil
+      return indexpr(property)
```

Comment:
This does not seem complete to me.

The `[]` do not differentiate index vs member expressions. For example `r["name"]` is a member expression as well as `r.name`. But `r[0]` or `r[i]` are index expressions. 

This should be possible to parse since if after the `[` a string literal follows then it is a member expression otherwise it is an index expression.

Reply:
`r["name"]` should be considered a member expression. I wasn't sure where this conversion should take place. I decided to put it in the semantic package. My reasoning was  so that the AST nodes represented specified lexical elements, whereas semantic nodes represented the meaning of those elements. But I'm totally fine having the conversion done inside of the parser if that makes more sense.